File: app/src/main/java/com/desuzed/clocknweather/WeatherFragment.java
package com.desuzed.clocknweather;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class WeatherFragment extends Fragment {
    private final String appId = "9154a207f4437c78b9bbfbffeeb98e1a";
    public static String BaseUrl = "https://api.openweathermap.org/";
    private TextView textView;

    public static WeatherFragment newInstance() {
        return new WeatherFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_weather, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        textView = view.findViewById(R.id.textView);
        Button b = view.findViewById(R.id.btn);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getWeather();
            }
        });
    }

    private void getWeather() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BASIC);
        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(logging)
                .build();

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BaseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build();
//VLD: 43.10562   131.87353
        //Galenki:
        WeatherService service = retrofit.create(WeatherService.class);
        Call<WeatherResponse> call = service.getCurrentWeatherData("44.04339", "131.7679", appId);
        call.enqueue(new Callback<WeatherResponse>() {
            @Override
            public void onResponse(Call<WeatherResponse> call, Response<WeatherResponse> response) {
                if (response.code() == 200) {
                    WeatherResponse wResponse = response.body();
                    float temp = Float.parseFloat(String.valueOf(wResponse.main.temp)) - 273;
                    String stringBuilder = "Country: " +
                            wResponse.sys.country +
                            "\n" +
                            "Temperature: " +
                            temp +
                            "\n" +
                            "Temperature(Min): " +
                            wResponse.main.temp_min +
                            "\n" +
                            "Temperature(Max): " +
                            wResponse.main.temp_max +
                            "\n" +
                            "Humidity: " +
                            wResponse.main.humidity +
                            "\n" +
                            "Pressure: " +
                            wResponse.main.pressure +
                            "\n" +
                            "Name: " + wResponse.name;

                    textView.setText(stringBuilder);
                }
            }

            @Override
            public void onFailure(Call<WeatherResponse> call, Throwable t) {

            }
        });
    }
}

File: app/src/main/java/com/desuzed/clocknweather/WeatherFragment.kt
package com.desuzed.clocknweather
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.desuzed.clocknweather.mvvm.WeatherViewModel
import com.desuzed.clocknweather.util.DailyAdapter
import com.desuzed.clocknweather.util.HourlyAdapter

class WeatherFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_weather, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val weatherViewModel: WeatherViewModel = ViewModelProvider(this,
            ViewModelProvider
                .AndroidViewModelFactory.getInstance(requireActivity().application))
            .get(WeatherViewModel::class.java)
        val tvCommonInfo = view.findViewById <TextView>(R.id.tvCommonInfo)
        val tvCurrentWeather = view.findViewById <TextView>(R.id.tvCurrentWeather)
        val etCity = view.findViewById<EditText>(R.id.etCity)
        val btnGetCityWeather = view.findViewById<Button>(R.id.bthGetWeather)
        val btnGpsWeather = view.findViewById<Button>(R.id.btnGpsWeather)
        btnGpsWeather.setOnClickListener {
//            if (etCity.text.toString().trim() == "") {
//                Toast.makeText(requireContext(), "Пустое поле", Toast.LENGTH_SHORT)
//            } else {
//                val city = etCity.text.toString().trim()
//                weatherViewModel.setCurrentWeatherByCity(city)
//            }
            weatherViewModel.getForecastOnecall("43.119809", "131.886924")
        }

        val rvDaily = view.findViewById<RecyclerView>(R.id.rvDaily)
        val rvHourly = view.findViewById<RecyclerView>(R.id.rvHourly)
        val lmDaily = LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
        val lmHourly = LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
        rvDaily.layoutManager = lmDaily
        rvHourly.layoutManager = lmHourly
        val hourlyAdapter = HourlyAdapter(ArrayList(), requireContext())
        val dailyAdapter = DailyAdapter(ArrayList(), requireContext())
        rvHourly.adapter = hourlyAdapter
        rvDaily.adapter = dailyAdapter

        weatherViewModel.weatherLiveData.observe(viewLifecycleOwner, {
           // textView.text = it
            tvCommonInfo.text = it.toString()
            tvCurrentWeather.text = it.current.toString()
            dailyAdapter.updateList(it.daily!!, it)
            hourlyAdapter.updateList(it.hourly!!, it)

        })

    }
//
    companion object {
        fun newInstance(): WeatherFragment {
            return WeatherFragment()
        }
    }
}